test/unmanaged.cpp

```diff
@@ -0,0 +1,191 @@
+
+    if (!basic_topic) {
+      std::cerr << "create_topic failed for '" << basic_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    test::BasicMessageTypeSupportImpl::_var_type complex_tsi = new test::BasicMessageTypeSupportImpl();
+
+    if (DDS::RETCODE_OK != complex_tsi->register_type(participant, "")) {
+      std::cerr << "register_type failed." << std::endl;
+      exit(1);
+    }
+
+    CORBA::String_var complex_type_name = complex_tsi->get_type_name();
+
+    DDS::TopicQos complex_topic_qos;
+    participant->get_default_topic_qos(complex_topic_qos);
+    DDS::Topic_var complex_topic =
+      participant->create_topic(complex_topic_name,
+                                complex_type_name,
+                                complex_topic_qos,
+                                DDS::TopicListener::_nil(),
+                                ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!complex_topic) {
+      std::cerr << "create_topic failed for '" << complex_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    DDS::PublisherQos pub_qos;
+    participant->get_default_publisher_qos(pub_qos);
+
+    DDS::Publisher_var pub =
+      participant->create_publisher(pub_qos, DDS::PublisherListener::_nil(),
+                                    ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+    if (!pub) {
+      std::cerr << "create_publisher failed." << std::endl;
+      exit(1);
+    }
+
+    DDS::DataWriterQos datawriter_qos;
+    pub->get_default_datawriter_qos(datawriter_qos);
+
+    DDS::DataWriter_var basic_dw =
+      pub->create_datawriter(basic_topic,
+                             datawriter_qos,
+                             DDS::DataWriterListener::_nil(),
+                             ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!basic_dw) {
+      std::cerr << "create_datawriter failed for '" << basic_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    test::BasicMessageDataWriter_var basic_message_dw =
+      test::BasicMessageDataWriter::_narrow(basic_dw);
+
+    if (!basic_message_dw) {
+      std::cerr << "test::BasicMessageDataWriter::_narrow() failed." << std::endl;
+      exit(1);
+    }
+
+    DDS::DataWriter_var complex_dw =
+      pub->create_datawriter(complex_topic,
+                             datawriter_qos,
+                             DDS::DataWriterListener::_nil(),
+                             ::OpenDDS::DCPS::DEFAULT_STATUS_MASK);
+
+    if (!complex_dw) {
+      std::cerr << "create_datawriter failed for '" << complex_topic_name << "'" << std::endl;
+      exit(1);
+    }
+
+    test::ComplexMessageDataWriter_var complex_message_dw =
+      test::ComplexMessageDataWriter::_narrow(complex_dw);
+
+    if (!complex_message_dw) {
+      std::cerr << "test::ComplexMessageDataWriter::_narrow() failed." << std::endl;
+      exit(1);
+    }
+
+    bool run = true;
+
+    const std::string id = std::string(argv[0]) + '-' + to_str(std::this_thread::get_id());
```

Comment:
```suggestion
    const std::string id = std::string(argv[0]) + '-' + std::to_string(std::this_thread::get_id());
```

Will `std::to_string` work here?

Reply:
I don't think so.  The `std::thread::id` type has an `operator<<` overload but not `to_string`.